src/main/java/com/ibm/eventstreams/kafkaconnect/plugins/xml/engines/StructToXmlBytes.java

```diff
@@ -233,17 +235,19 @@ private void processStruct(Document doc, Element parentElement, Struct source) {
 
                 case BYTES:
                     final Element bytesElement = doc.createElement(field.name());
-                    if (source.getBytes(field.name()) != null) {
-                        final byte[] bytes = source.getBytes(field.name());
-
+                    final byte[] bytes =
+                            Decimal.LOGICAL_NAME.equals(fieldSchema.name()) ?
+                                    Decimal.fromLogical(fieldSchema, new BigDecimal(String.valueOf(source.get(field.name())))) :
+                                    source.getBytes(field.name());
+                    if (bytes != null) {
```

Comment:
Under what situations could `bytes` be null?

Reply:
Since that  condition was already existing just kept  as it is.  I have not tested out the scenarios where it could be null.